official/recommendation/ncf_test.py

```diff
@@ -23,10 +23,56 @@
 import numpy as np
 import tensorflow as tf
 
+from official.recommendation import constants as rconst
+from official.recommendation import neumf_model
 from official.recommendation import ncf_main
+from official.recommendation import stat_utils
+
+
+NUM_TRAIN_NEG = 4
 
 
 class NcfTest(tf.test.TestCase):
+  def setUp(self):
+    self.top_k_old = rconst.TOP_K
+    self.num_eval_negatives_old = rconst.NUM_EVAL_NEGATIVES
+    rconst.NUM_EVAL_NEGATIVES = 2
+
+  def tearDown(self):
+    rconst.NUM_EVAL_NEGATIVES = self.num_eval_negatives_old
+    rconst.TOP_K = self.top_k_old
+
+  def get_hit_rate_and_ndcg(self, predicted_scores_by_user, items_by_user,
+                            top_k=rconst.TOP_K, match_mlperf=False):
+    rconst.TOP_K = top_k
+    rconst.NUM_EVAL_NEGATIVES = predicted_scores_by_user.shape[1] - 1
+
+    g = tf.Graph()
+    with g.as_default():
```

Comment:
`with tf.Graph().as_default() as g`

Reply:
I'm not overly keen on keeping the context manager variable to use to pass to the session later.